aloha-h2o/src/main/scala/com/eharmony/aloha/models/h2o/compiler/compiler.scala

```diff
@@ -0,0 +1,116 @@
+package com.eharmony.aloha.models.h2o.compiler
+
+import java.io.File
+import java.net.URLClassLoader
+import java.nio.charset.Charset
+import java.util.Locale
+import javax.tools.JavaCompiler.CompilationTask
+import javax.tools.{DiagnosticCollector, JavaFileObject, ToolProvider}
+
+import com.eharmony.aloha.io.{AlohaReadable, ReadableByString}
+import com.eharmony.aloha.reflect.{RefInfo, RefInfoOps}
+import com.eharmony.aloha.util.Logging
+import org.apache.commons.io.FileUtils
+
+import scala.collection.JavaConversions.{asJavaIterable, collectionAsScalaIterable}
+import scala.util.{Failure, Success, Try}
+
+
+/**
+ * A compiler that compiles and instantiates the code.
+ * @param classDir If supplied, an attempt to create the directory but no attempt to delete the
+ *                 directory will be made.
+ * @param locale A Locale
+ * @param charSet A CharSet
+ * @param riB reflection info about the output type B.
+ * @tparam B the output type.  This should be an interface or base class that the compiled class extends.
+ */
+private[h2o] class Compiler[B](classDir: Option[File] = None,
+                               locale: Locale   = Locale.getDefault,
+                               charSet: Charset = Charset.defaultCharset)
+                              (implicit riB: RefInfo[B])
+extends AlohaReadable[Try[B]]
+   with ReadableByString[Try[B]] {
+
+  /** Read from a String.
+    * @param code a String to read.
+    * @return the result
+    */
+  override def fromString(code: String): Try[B] = {
+    val diagnosticCollector = new DiagnosticCollector[JavaFileObject]
+
+    for {
+      dirAndDelete         <- getCompileDir(classDir)
+      compileDir           = dirAndDelete._1  // Avoids: `withFilter' method does not yet exist on
+      delete               = dirAndDelete._2  //         Try[(File, Boolean)], using `filter' method instead
+      _                    <- mkDir(compileDir)
+      compilationUnit      <- getCompilationUnit(code)
+      compileTask          <- getCompileTask(compilationUnit, compileDir, diagnosticCollector)
+      diagnostics          <- compile(compileTask, diagnosticCollector)
+      untypedInstance      <- instantiate(compilationUnit, compileDir)
+      instance             <- cast(untypedInstance)
+      _                    <- rmDir(compileDir, delete)
+    } yield instance
+  }
+
+  def getCompileDir(dir: Option[File]) = dir.map(d => Try{(d, false)}) getOrElse Compiler.tmpDir.map(d => (d, true))
+
+  def getCompileTask(compilationUnit: InMemoryJavaSource[B],
+                     compileDir: File,
+                     diagnosticCollector: DiagnosticCollector[JavaFileObject]) = Try {
+    val locations = asJavaIterable(Seq("-d", compileDir.getCanonicalPath))
+    val compiler = ToolProvider.getSystemJavaCompiler
```

Comment:
This would seem to imply that javac must be on the machine running the model.  Is this so?